recipes/mpi-serial/meta.yaml

```diff
@@ -0,0 +1,68 @@
+{% set name = "mpi-serial" %}
+{% set version = "2.3.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/MCSclimate/mpi-serial/archive/refs/tags/MPIserial_{{ version }}.tar.gz
+  sha256: cc55e6bf0ae5e1d93aafa31ba91bfc13e896642a511c3101695ea05eccf97988
+  patches:
+    - 0001-Rearrange-code-and-other-minor-fixes-needed-for-comp.patch
+    - fix-makefile.patch
+
+build:
+  number: 0
+  skip: true  # [win]
```

Comment:
Add `run_exports`?

Reply:
Indeed. I don't know the package very well.  Based on the recent tags, I'd be inclined to pin it as:
```
  run_exports:
    - {{ pin_subpackage('mpi_serial', max_pin='x.x') }} {{ build_pin }}
```

@rljacob, can you give us a sense of the versioning for `mpi-serial`?  Will there be any API/ABI changes with minor version updates?

Reply:
This would mean that, if we build with `mpi-serial` 2.3.0 (the latest tag at the moment), we would not need to rebuild (i.e. have a conda-forge migrator) for a bug fix update to 2.3.1 but would for a new minor release to 2.4.0.  If we hear from @rljacob that any 2.x.x tag does not have any breaking API/ABI changes, we could loosen to `max_pin='x'`.

Reply:
I have added the `run_exports` as suggested by @xylar (I am not knowledgable enough to have an opinion on my own here). Nevertheless, my previous question stands, so would be good to address in review.

Do we also need a global pin or somesuch?

Reply:
@zklaus

> What impact does it have that this is only a static library and that the API is essentially dictated by a very stable standard (modulo addition of newly supported functions)?

Right, the fact that it's a static library probably makes the whole discussion moot, since what you get a runtime is exactly what went in a build time.

The stable API makes me think that a pin of `x` rather than `x.x` would almost certainly be fine (even if we had shared libraries).  But I also think updates are so infrequent that it probably doesn't matter very much.

> Do we also need a global pin or somesuch?

I'm not sure I follow what this is referring to but my first guess is you mean the equivalent of:
https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml#L570-L571

I do think we need that once this goes in.  And I think it would get added to `recipe/conda_build_config.yaml` in a PR like this one that added `mpich` (among many others):
https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/270

Reply:
@zklaus, let's see what @isuruf has to say.  Of course, ideally we wouldn't be building a static library but that would be a considerably bigger project.  But I do agree that it's not obvious to me that `run_exports` is a useful thing to have for a static library.

Reply:
I think it's time to move forward. I vote to keep the `run_exports` and to work towards a dynamic version of `mpi_serial`.